### PR TITLE
test(docs): lint MDX

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

**Issue**: Without Linters, it's 1) easy to introduce bug without known (e.g. forget to import variables https://github.com/aws-amplify/amplify-ui/pull/3339), 2) code issues in code blocks 3) code style inconsistent

**Fix**: The goal of this PR is to 

1. Lint all MDX fiiles under `docs/`
2. Add linter to prevent issue at build time

Lint config references:

- [eslint-mdx](https://github.com/remarkjs/remark-lint)
- [remarklint](https://github.com/remarkjs/remark-lint)

#### Description of how you validated changes

- Tested on localhost
- ToC https://docs-lint-mdx.da9impzjamkt4.amplifyapp.com/
- Ran `build` for both success and failed lint.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/src/components/ThemeExample.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Link, View } from '@aws-amplify/ui-react';
```
Comment (0618) [inline docs/src/components/ThemeExample.tsx:1]: This file was converted from docs/src/components/ThemeExample.mdx because

1. {props.component} is hard to lint in mdx
2. we don't have other mdx component in docs/

File: docs/src/pages/[platform]/components/alert/react.mdx [hunk 2]
```diff
@@ -215,6 +218,7 @@ _Using a class selector:_
   <ExampleCode>
 
 ```jsx
+import { Alert } from '@aws-amplify/ui-react';
```
Comment (0618) [inline docs/src/pages/[platform]/components/alert/react.mdx:221]: Have to import all the components in code block

File: docs/.eslintrc.json [hunk 3]
```diff
@@ -1,3 +1,18 @@
...
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "overrides": [
+    {
+      "files": ["*.angular.mdx"],
```
Comment (0618) [inline docs/.eslintrc.json:14]: override Angular files because the code block would have the following error

> Support for the experimental syntax 'decorators-legacy' isn't currently enabled (6:1):

File: docs/src/pages/[platform]/guides/css-in-js/web.mdx [hunk 4]
```diff
@@ -151,22 +153,27 @@ These are styling props that can be used directly on an Amplify component which
...
+import { Text, View } from '@aws-amplify/ui-react';
 
 const useStyles = createUseStyles({
   styledText: {
     color: 'blue',
   },
 });
 
-const classes = useStyles();
+export function JSSText() {
```
Comment (0618) [inline docs/src/pages/[platform]/guides/css-in-js/web.mdx:164]: Have to wrap React Hook in a function

> React Hook "useStyles" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.eslint_[react-hooks/rules-of-hooks](https://reactjs.org/docs/hooks-rules.html)_ (https://reactjs.org/docs/hooks-rules.html)

File: docs/src/pages/[platform]/theming/default-theme/tokenUsage.react.mdx [hunk 5]
```diff
@@ -10,10 +10,10 @@ const MyComponent = () => {
 };
 ```
 
-Or we can use the design token to [create a custom theme](/react/theming). 
+Or we can use the design token to [create a custom theme](/react/theming).
 
 ```javascript
-const theme {
+const theme = {
```
Comment (0618) [inline docs/src/pages/[platform]/theming/default-theme/tokenUsage.react.mdx:16]: I'm glad the linter is able to find it

File: docs/src/pages/[platform]/theming/responsive/web.mdx [hunk 6]
```diff
@@ -5,7 +5,7 @@ are used for all breakpoints.
 
 ### Breakpoints
 
-```js
+```css
```
Comment (0618) [inline docs/src/pages/[platform]/theming/responsive/web.mdx:8]: Another issue the linter caught!

File: docs/package.json [hunk 7]
```diff
@@ -9,9 +9,10 @@
     "postflutter:build": "yarn flutter:copy",
     "flutter:clean": "rm -rf public/flutter/authenticator",
     "flutter:copy": "cp -r ./flutter/authenticator/build/web public/flutter/authenticator",
-    "build": "yarn build:props-tables && next build",
+    "build": "yarn build:props-tables && yarn lint && next build",
```
Comment (wlee221) [inline docs/package.json:12]: Should `lint` be in `build`? IMO it should be separate step, and we would explicitly call this in CI/CD or on Amplify Hosting.

File: docs/package.json [hunk 8]
```diff
@@ -9,9 +9,10 @@
...
     "build:props-tables": "node --require esbuild-register ./scripts/generate-props-tables-data.ts",
-    "lint": "next lint",
+    "lint": "next lint && eslint . --ext ts,tsx,md,mdx",
+    "lint:fix": "eslint . --ext ts,tsx,md,mdx --fix",
```
Comment (wlee221) [inline docs/package.json:15]: What do you think about removing `lint:fix` and instead we can do `yarn lint --fix`?

File: .vscode/settings.json [hunk 9]
```diff
@@ -8,5 +8,16 @@
...
+    "extensions": [".js", ".jsx", ".md", ".mdx", ".ts", ".tsx"]
+  },
+  "eslint.validate": [
+    "markdown",
+    "mdx",
+    "javascript",
+    "javascriptreact",
+    "typescript",
+    "typescriptreact"
+  ]
```
Comment (calebpollman) [inline .vscode/settings.json:22]: Why is this here?

**Comment by 0618** [comment]
Will break down this PR to several smaller ones.

File: .vscode/settings.json (hunk 9 above)
Comment (0618) [reply .vscode/settings.json:22]: wait a sec...

Comment (0618) [reply .vscode/settings.json:22]: cc. @calebpollman in case you don't get the notification

File: docs/package.json (hunk 8 above)
Comment (0618) [reply docs/package.json:15]: Good call. I'm thinking of changing it to a pre-commit instead in a future PR

File: docs/package.json (hunk 7 above)
Comment (0618) [reply docs/package.json:12]: Good call. I'm thinking of changing it to a pre-commit instead in a future PR
